main.js

```diff
@@ -0,0 +1,284 @@
+                let obj = {
+                    news: {posts: news.sort((a,b) => Math.sign(b.date_t - a.date_t))}
+                };
+                let finalResult = Object.assign(obj, results)
+                resolve(finalResult);
+            })
+        })
+    });
+}
+
+/**
+ * Writes news.html and news articles from news object
+ * @param {Object} results object including globalContext, news, and template objects
+ * @param {String} dirname folder to write news to
+ */
+function writeNews(results, dirname='news') {
+    // huh, stuff gets real simple when it's sync 🤔
+    fse.mkdirpSync(path.join(outDir, dirname));
+
+    let promises = results.news.posts.map(newsObj =>
+        fse.writeFile(path.join(outDir, newsObj.url), 
+            results.wrapperTemplate(
+                Object.assign({
+                    body: results.articleTemplate({
+                        title: newsObj.title,
+                        date: newsObj.date,
+                        body: newsObj.body
+                    }),
+                    title: newsObj.title,                    
+                }, results.globalContext)
+            )
+        )
+    )
+    promises.push(fse.writeFile(path.join(outDir, 'news.html'),
+        results.wrapperTemplate(Object.assign({
+            title: "News",
+            body: results.newslistTemplate(results.news)
+        }, results.globalContext))
+    ))
+    return Promise.all(promises);
+}
+
+/**
+ * Writes the index page
+ * @param {Object} results object including globalContext, news, and template objects.
+ */
+function writeIndex(results) {
+    return fse.writeFile(path.join(outDir, 'index.html'), results.wrapperTemplate(
+        Object.assign({
+            body: results.indexTemplate(Object.assign({
+                newslist: results.newslistTemplate({
+                    posts: results.news.posts.slice(0,5),
+                    link: true
+                }) // first 5 news articles (if present)
+            }, results.globalContext))
+            // no title, handled by wrapper.handlebars
+        }, results.globalContext)
+    ));
+}
+
+/**
+ * @returns {Promise<Object>} Resolves to the global context object
+ */
+function getGlobalContext() {
+    return fse.readFile('templates/context.yaml')
+        .then(contents => {
+            /**
+             * {Buffer} contents: the contents of the YAML file
+             */
+            return yaml.safeLoad(contents.toString('UTF-8'));
+        })
+}
+
+// Main stuff here.
+let p_mkOutDir = fse.mkdirp(outDir)
+    .then(() => fse.emptyDir(outDir));
+
+let p_copyStatic = fse.copy('static', path.join(outDir, 'static'))
+    .then(console.log("Copied static files!"));
+     
+    p_mkOutDir.then(() => p_copyStatic);
+
+    let hljsStylesheet = 'solarized-light';
+    p_copyStatic.then(() => fse.copyFile(`node_modules/highlight.js/styles/${hljsStylesheet}.css`, path.join(outDir, 'static', 'highlight.css')));
```

Comment:
Special static copies should probably be stored in a config file, though that could be implemented at a later date.

Reply:
What do you mean by special static copies? The copy operation on L243?

Reply:
It's nonstandard in that it's not `static/`, however I regard it as part of the Markdown build process, specific to how this system builds the site. The content in `static/` should be for the most part separable from the build system. In addition, this is the only occurrence of nonstandard copying, and it is from a well-defined location. If there's more like this added later, then perhaps a config could be considered, but I think this instance is acceptable.

Reply:
Hmm this may be a case of my prematurely generalising the code. If you think this is the only special copy into static ever likely to be needed then sure, I'm happy with it as is, I'd just probably have generalised. Fine to leave as is though.